syntax = "proto3";
import "google/protobuf/timestamp.proto";
package gcs_system.Services;

service DroneStatusUpdate {
  rpc UpdateDroneStatus (UpdateDroneStatusPayload) returns (StatusResponse);
}

message UpdateDroneStatusPayload {
  repeated GrpcDroneStatus droneStatuses = 1;
}

message GrpcDroneStatus {
  optional string DroneId = 1;
  optional string FlightId = 2;
  optional bool IsOnline = 3;
  optional bool IsLanded = 4;
  optional string ControllStt = 5;
  optional GrpcDroneStt DroneStt = 6;
  optional GrpcSensorData SensorData = 7;
}

message GrpcDroneStt {
  optional float PowerV = 1;
  optional sint32 BatteryStt = 2;
  optional string GpsStt = 3;
  optional double TempC = 4;
  optional double Lat = 5;
  optional double Lon = 6;
  optional double Alt = 7;
  optional double GlobalAlt = 8;
  optional double Roll = 9;
  optional double Pitch = 10;
  optional int32 Head = 11;
  optional float Speed = 12;
  optional string HoverStt = 13;
  optional double HDOP = 14;
  optional int32 SatellitesCount = 15;
  optional int32 FlightMode = 16;
}

message GrpcSensorData {
  optional float roll_ATTITUDE = 1;
  optional float pitch_ATTITUDE = 2;
  optional float yaw_ATTITUDE = 3;
  optional int32 xacc_RAW_IMU = 4;
  optional int32 yacc_RAW_IMU = 5;
  optional int32 zacc_RAW_IMU = 6;
  optional int32 xgyro_RAW_IMU = 7;
  optional int32 ygyro_RAW_IMU = 8;
  optional int32 zgyro_RAW_IMU = 9;
  optional int32 xmag_RAW_IMU = 10;
  optional int32 ymag_RAW_IMU = 11;
  optional int32 zmag_RAW_IMU = 12;
  optional float vibration_x_VIBRATION = 13;
  optional float vibration_y_VIBRATION = 14;
  optional float vibration_z_VIBRATION = 15;
  optional float accel_cal_x_SENSOR_OFFSETS = 16;
  optional float accel_cal_y_SENSOR_OFFSETS = 17;
  optional float accel_cal_z_SENSOR_OFFSETS = 18;
  optional int32 mag_ofs_x_SENSOR_OFFSETS = 19;
  optional int32 mag_ofs_y_SENSOR_OFFSETS = 20;
  optional int32 vx_GLOBAL_POSITION_INT = 21;
  optional int32 vy_GLOBAL_POSITION_INT = 22;
  optional float x_LOCAL_POSITION_NED = 23;
  optional float vx_LOCAL_POSITION_NED = 24;
  optional float vy_LOCAL_POSITION_NED = 25;
  optional float nav_pitch_NAV_CONTROLLER_OUTPUT = 26;
  optional int32 nav_bearing_NAV_CONTROLLER_OUTPUT = 27;
  optional uint32 servo3_raw_SERVO_OUTPUT_RAW = 28;
  optional uint32 servo8_raw_SERVO_OUTPUT_RAW = 29;
  optional float groundspeed_VFR_HUD = 30;
  optional float airspeed_VFR_HUD = 31;
  optional float press_abs_SCALED_PRESSURE = 32;
  optional uint32 Vservo_POWER_STATUS = 33;
  optional double voltages1_BATTERY_STATUS = 34;
  optional int32 chancount_RC_CHANNELS = 35;
  optional uint32 chan12_raw_RC_CHANNELS = 36;
  optional uint32 chan13_raw_RC_CHANNELS = 37;
  optional uint32 chan14_raw_RC_CHANNELS = 38;
  optional uint32 chan15_raw_RC_CHANNELS = 39;
  optional uint32 chan16_raw_RC_CHANNELS = 40;
}

message StatusResponse {
  string DroneId = 1;
  GrpcPredictData PredictData = 2;
  GrpcWarningData WarningData = 3;
}

message GrpcPredictData {
  double rollATTITUDE_PREDICT = 1;
  double yawATTITUDE_PREDICT = 2;
  double pitchATTITUDE_PREDICT = 3;
  double xaccRAWIMU_PREDICT = 4;
  double yaccRAWIMU_PREDICT = 5;
  double zaccRAWIMU_PREDICT = 6;
  double xgyroRAWIMU_PREDICT = 7;
  double ygyroRAWIMU_PREDICT = 8;
  double zgyroRAWIMU_PREDICT = 9;
  double xmagRAWIMU_PREDICT = 10;
  double ymagRAWIMU_PREDICT = 11;
  double zmagRAWIMU_PREDICT = 12;
  double vibrationXVIBRATION_PREDICT = 13;
  double vibrationYVIBRATION_PREDICT = 14;
  double vibrationZVIBRATION_PREDICT= 15;
}

message GrpcWarningData  {
  int32 warning_count = 1;
  bool rollATTITUDE_WARNING = 2;
  bool pitchATTITUDE_WARNING = 3;
  bool yawATTITUDE_WARNING = 4;
  bool xaccRAWIMU_WARNING = 5;
  bool yaccRAWIMU_WARNING = 6;
  bool zaccRAWIMU_WARNING = 7;
  bool xgyroRAWIMU_WARNING = 8;
  bool ygyroRAWIMU_WARNING = 9;
  bool zgyroRAWIMU_WARNING = 10;
  bool xmagRAWIMU_WARNING = 11;
  bool ymagRAWIMU_WARNING = 12;
  bool zmagRAWIMU_WARNING = 13;
  bool vibrationXVIBRATION_WARNING = 14;
  bool vibrationYVIBRATION_WARNING = 15;
  bool vibrationZVIBRATION_WARNING = 16;
}

